src/domain-analysis.js

```diff
@@ -0,0 +1,50 @@
+/* This file contains the code for analyzing number of blog posts from different domains.
+ * The module receives an array of blog urls and returns an array of json object in format
+ * {name: domainName, count: number of frequency}
+*/
+
+//  import url library
+const urls = require('url');
+
+/*  getDomain() is used to get the domain from hostname without username.
+*   It will split the hostname that pass in, if there is only two parts,
+*   then it will return the hostname;
+*   if not, it will remove the first part.
+*/
+function getDomain(hostname) {
+  const hostNameCounter = hostname.split('.').length;
+  const domain = hostname.substring(hostname.indexOf('.') + 1);
+  if (hostNameCounter === 2) {
+    return hostname;
+  }
+  return domain;
+}
+
+/* createDomainObject() will create an object if it's not exist */
+function createDomainObject(domainName) {
+  return { name: domainName, count: 1 };
+}
+
+/* it will return object array: {name: domainName, count: numberOfDomain} */
+module.exports.countDomain = function (feedUrls) {
```

Comment:
"Domain" is very vague as this could apply to any domain name. Since you are trying to count how many people are using certain blogging services, perhaps a name that reflects this is more appropriate